src/miner.cpp

```diff
@@ -573,9 +551,14 @@ void GenerateBitcoins(bool fGenerate, CWallet* pwallet, int nThreads)
     if (nThreads == 0 || !fGenerate)
         return;
 
+    CScript coinbaseScript;
+    GetMainSignals().ScriptForMining(coinbaseScript);
```

Comment:
@luke-jr: if the signal listener don't provide a script, a exception get thrown at L559. If there are multiple listeners, one listener can detect this by reading out the passed-by-reference `coinbaseScript` (check if it's empty, etc.) and skip the creation/generation of another `CScript` or do some other clever things. IMO this design is acceptable for the internal miner/wallet decoupling.